scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java

```diff
@@ -766,6 +775,274 @@ public DatasetId storeSimulationResults(final PlanMetadata planMetadata,
+    for(final var unfinishedActivityJson: unfinishedActivitiesJson){
+      final var activityAttributes = activityAttributesP.parse(unfinishedActivityJson.asJsonObject().getJsonObject("attributes")).getSuccessOrThrow();
+      SimulatedActivityId parentId = null;
+      if(!unfinishedActivityJson.asJsonObject().isNull("parent_id")){
+        parentId = new SimulatedActivityId(unfinishedActivityJson.asJsonObject().getJsonNumber("parent_id").longValue());
+      }
+      final var activityType = unfinishedActivityJson.asJsonObject().getJsonString("type").getString();
+      final var start = instantFromStart(simulationStart,
+          durationFromPGInterval(unfinishedActivityJson.asJsonObject().getJsonString("start_offset").getString()));
+      final var id = new SimulatedActivityId(unfinishedActivityJson.asJsonObject().getJsonNumber("id").longValue());
+      Optional<ActivityDirectiveId> actDirectiveId = Optional.empty();
+      if(activityAttributes.directiveId().isPresent()){
+        actDirectiveId = Optional.of(new ActivityDirectiveId(activityAttributes.directiveId().get()));
+      }
+      final var unfinishedActivity = new UnfinishedActivity(
+          activityType,
+          activityAttributes.arguments(),
+          start,
+          parentId,
+          List.of(),
+          actDirectiveId
+      );
+      unfinishedActivities.put(id, unfinishedActivity);
+    }
+    return unfinishedActivities;
+  }
+
+  private record Profiles(
+      Map<String, Pair<ValueSchema, List<ProfileSegment<RealDynamics>>>> realProfiles,
+      Map<String, Pair<ValueSchema, List<ProfileSegment<SerializedValue>>>> discreteProfiles
+  ){}
+
+  private Profiles parseProfiles(JsonArray dataset){
+    Map<String, Pair<ValueSchema, List<ProfileSegment<RealDynamics>>>> realProfiles = new HashMap<>();
+    Map<String, Pair<ValueSchema, List<ProfileSegment<SerializedValue>>>> discreteProfiles = new HashMap<>();
+    for(final var profile:dataset){
+      final var duration = durationFromPGInterval(profile.asJsonObject().getString("duration"));
+      final var name = profile.asJsonObject().getString("name");
+      final var type = profile.asJsonObject().getJsonObject("type");
+      final var typetype = type.getString("type");
+      final boolean isReal = typetype.equals("real");
+      if(isReal){
+        final var realProfile = parseProfile(profile.asJsonObject(), duration, realDynamicsP);
+        realProfiles.put(name, realProfile);
+      } else {
+        final var discreteProfile = parseProfile(profile.asJsonObject(), duration, serializedValueP);
+        discreteProfiles.put(name, discreteProfile);
+      }
+    }
+    return new Profiles(realProfiles, discreteProfiles);
+  }
+
+  public <Dynamics> Pair<ValueSchema, List<ProfileSegment<Dynamics>>> parseProfile(JsonObject profile, Duration profileExtent, JsonParser<Dynamics> dynamicsParser){
+    final var name = stringP.parse(profile.getJsonString("name"));
```

Comment:
Very small method signature refinement:
```suggestion
  public <Dynamics> Pair<ValueSchema, List<ProfileSegment<Dynamics>>> parseProfile(JsonObject profile, JsonParser<Dynamics> dynamicsParser){
    final var profileExtent = durationFromPGInterval(profile.asJsonObject().getString("duration"));
```
With this we also don't need to grab the `duration` in `parseProfiles` (L916)

Reply:
done.